#14581: Remove deprecated UMD API host_addr_params

blozano-tt opened the pull request:
### Ticket
Closes #14581 
#596 

### Problem description
Circular interaction with UMD has been fixed. No need to do this. UMD will self initialize these parameters based on detected ARCH.

### What's changed
Removed usage of deprecated API

### Checklist
https://github.com/tenstorrent/tt-metal/actions/runs/11695072302
- [x] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [x] New/Existing tests provide coverage for changes

blozano-tt committed:
#14581: Remove usage of deprecated UMD API

afuller-TT reviewed (approved):
I love me a PR that's nothin' but red!  🥳

blozano-tt committed:
Update CMakeLists.txt

Put the include paths back to be safe for now
Still need eth_interface.h for now.
